jsonalchemy/reader.py

```diff
@@ -422,8 +422,11 @@ def set_default_value(field, schema):
                 self._json.__setitem__(field_name, value, extend=False,
                                        exclude=['decorators', 'extensions'])
             else:
-                old_value = self._json.__getitem__(
-                    field_name, exclude=['decorators', 'extensions'])
+                try:
+                    old_value = self._json.__getitem__(
+                        field_name, exclude=['decorators', 'extensions'])
+                except KeyError:
+                    old_value = value
```

Comment:
I wrote you some time ago on `gitter` what the error was about. As you couldn't explain to me what this code was supposed to do, I just caught the error that was produced by this line (I think that the fact that error is thrown is wrong), and secured it. IMO in the example given by me, the `if value is not None:` should not be passed, but it is. Unfortunately this is only a guess from me, I don't want to follow all the changes on the dictionary used in the module.

And I will add here my message from gitter

```
There is a function in reader.py called set_default_value
It creates a default value in case no other is provided.
In case of dictionary it recursively put Nones in every field
So, in this case, it returns a dictionary
Then, the _set_default_value encounters if value is not None, and the not provided value passes through this
Every other fields, that is not provided and is not a dictionary doesn't go there as set_default_value returns None
Then, in case of the dictionary, the wrapper doesn't have this key set in self._dict, which apparently happens only when the default value is provided.
```